src/main/java/com/github/mizool/jediscache/JedisCache.java

```diff
@@ -0,0 +1,353 @@
+    }
+
+    @Override
+    public V get(K key)
+    {
+        verifyOpen();
+        try (Jedis jedis = jedisPool.getResource())
+        {
+            return valueConverter.toValue(jedis.hget(jedisCacheName, keyConverter.fromKey(key)));
+        }
+    }
+
+    @Override
+    public Map<K, V> getAll(Set<? extends K> keys)
+    {
+        verifyOpen();
+        Map<byte[], byte[]> allEntries;
+        try (Jedis jedis = jedisPool.getResource())
+        {
+            allEntries = jedis.hgetAll(jedisCacheName);
+        }
+        ImmutableMap.Builder<K, V> builder = ImmutableMap.builder();
+        keys.forEach(key -> {
+            byte[] serializedKey = keyConverter.fromKey(key);
+            if (allEntries.containsKey(serializedKey))
+            {
+                builder.put(key, valueConverter.toValue(allEntries.get(serializedKey)));
+            }
+        });
+        return builder.build();
+    }
+
+    @Override
+    public boolean containsKey(K key)
+    {
+        verifyOpen();
+        try (Jedis jedis = jedisPool.getResource())
+        {
+            return jedis.hexists(jedisCacheName, keyConverter.fromKey(key));
+        }
+    }
+
+    @Override
+    public void loadAll(
+        Set<? extends K> keys, boolean replaceExistingValues, CompletionListener completionListener)
+    {
+        throw new UnsupportedOperationException("listener support is not implemented");
+    }
+
+    @Override
+    public void put(K key, V value)
+    {
+        verifyOpen();
+        try (Jedis jedis = jedisPool.getResource())
+        {
+            jedis.hset(jedisCacheName, keyConverter.fromKey(key), valueConverter.fromValue(value));
+        }
+    }
+
+    @Override
+    public V getAndPut(K key, V value)
+    {
+        verifyOpen();
+        V result = get(key);
+        put(key, value);
+        return result;
+    }
+
+    @Override
+    public void putAll(Map<? extends K, ? extends V> map)
+    {
+        map.forEach(this::put);
+    }
+
+    @Override
+    public boolean putIfAbsent(K key, V value)
+    {
+        verifyOpen();
+        try (Jedis jedis = jedisPool.getResource())
+        {
+            return jedis.hsetnx(jedisCacheName, keyConverter.fromKey(key), valueConverter.fromValue(value)) == 1;
+        }
+    }
+
+    @Override
+    public boolean remove(K key)
+    {
+        verifyOpen();
+        try (Jedis jedis = jedisPool.getResource())
+        {
+            return jedis.hdel(jedisCacheName, keyConverter.fromKey(key)) == 1;
+        }
+    }
+
+    @Override
+    public boolean remove(K key, V oldValue)
+    {
+        return get(key).equals(oldValue) && remove(key);
+    }
+
+    @Override
+    public V getAndRemove(K key)
+    {
+        V result = get(key);
+        remove(key);
+        return result;
+    }
+
+    @Override
+    public boolean replace(K key, V oldValue, V newValue)
+    {
+        boolean result = containsKey(key);
+        if (result)
+        {
+            put(key, newValue);
```

Comment:
Yet another race condition

Reply:
... and a bug: you don't use the given `oldValue`, violating [the spec](https://static.javadoc.io/javax.cache/cache-api/1.0.0/javax/cache/Cache.html#replace(K,%20V,%20V))